docs/ci/demo

```diff
@@ -210,6 +210,15 @@ popper cleanup
 ! popper stages mypipetwo | grep teardown
 ! popper stages mypipetwo | grep post-run
 
+# popper matrix subcommand
+init_test
+popper init mypipeone
+popper matrix mypipeone --add key1=val1 --add key2=val2
+popper matrix mypipeone | grep -q key1
+popper matrix mypipeone | grep -q key2
+popper matrix mypipeone --rm key1
+! popper matrix mypipeone | grep -q key1
```

Comment:
this will not work as intended. For example:

```bash
popper matrix mypipeone --add key1=val1 --add key2=val2
popper matrix mypipeone --add key1=val1 --add key2=other
```

the above defines _two_ elements, so when we remove an entry, we remove it by providing all the key-value pairs in that item. Hmmm, maybe we should rename the `matrix` subcommand to `vars`?

Reply:
Becuase of the way dictionaries work in python, the above will replace key2 with the value other. So, there will only be one instance of key2.